### PR TITLE
run tests in GitHub actions

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - ci/test/nightly
```
Comment (mzabaluev) [inline .github/workflows/test.yml:7]: No need for this match, that branch is used to test release builds.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    needs: cache_info
+    # Caches on Windows and Unix do not interop:
+    # https://github.com/actions/cache/issues/330#issuecomment-637701649
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      RUSTFLAGS: -D warnings
+      RUST_BACKTRACE: 1
```
Comment (mzabaluev) [inline .github/workflows/test.yml:53]: These environment variables look out of place here. I'd expect them to be in the `tests` job, however that one does not set them.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        flags: ['', '--release']
+        toolchain: [stable, nightly]
+    steps:
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+          default: true
+          components: rustfmt
```
Comment (mzabaluev) [inline .github/workflows/test.yml:96]: rustfmt shouldn't be needed if you build jormungandr with `--no-default-features`, but then you can only use the `--manifest-path` option, so maybe it's not worth bothering with.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  tests:
+    name: Tests
+    needs: [cache_info, update_deps]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        flags: ['', '--release']
+        toolchain: [stable, nightly]
```
Comment (mzabaluev) [inline .github/workflows/test.yml:89]: To reduce the number of jobs, maybe only test with `nightly` toolchain on `ubuntu-latest` in debug profile.
We need it as a canary to catch unlikely future regressions.

File: .github/workflows/test.yml (hunk 2 above)
Comment (eugene-babichenko) [reply .github/workflows/test.yml:53]: Ouch, you're right, thanks for noticing that!

**Review by mzabaluev — APPROVED** [review]

**Comment by mzabaluev** [comment]
Looks good. The lints job invokes a newer clippy than the CircleCI check, so we've got a few more failures.

**Pull request merged**